frontends/election-manager/src/components/print_all_ballots_button.tsx

```diff
@@ -0,0 +1,362 @@
+
+import {
+  assert,
+  throwIllegalValue,
+  BallotStyleData,
+  sleep,
+} from '@votingworks/utils';
+import {
+  Modal,
+  Prose,
+  isAdminAuth,
+  isSuperadminAuth,
+  useCancelablePromise,
+  Loading,
+  useLock,
+} from '@votingworks/ui';
+import { LogEventId } from '@votingworks/logging';
+import pluralize from 'pluralize';
+import {
+  getBallotStylesData,
+  sortBallotStyleDataByPrecinct,
+} from '../utils/election';
+
+import { AppContext } from '../contexts/app_context';
+import { HandMarkedPaperBallot } from './hand_marked_paper_ballot';
+import { Button } from './button';
+import { LinkButton } from './link_button';
+
+import {
+  BallotMode,
+  ballotModeToReadableString,
+  PrintableBallotType,
+} from '../config/types';
+import { BallotTypeToggle } from './ballot_type_toggle';
+import { BallotModeToggle } from './ballot_mode_toggle';
+import { PrintableArea } from './printable_area';
+import { DEFAULT_LOCALE } from '../config/globals';
+import { BallotCopiesInput } from './ballot_copies_input';
+import { PrintBallotButtonText } from './print_ballot_button_text';
+
+export const PRINTER_WARMUP_TIME = 8300;
+export const TWO_SIDED_PRINT_TIME = 3300;
+export const PRINTER_COOLDOWN_TIME = 5000;
+
+const CenteredOptions = styled.div`
+  text-align: center;
+  & > :last-child {
+    margin-bottom: 0;
+  }
+`;
+
+type ModalState = 'no-printer' | 'options' | 'printing';
+
+const defaultBallotLocales: BallotLocales = { primary: DEFAULT_LOCALE };
+
+interface Props {
+  draftMode?: boolean;
+}
+
+export function PrintAllBallotsButton({ draftMode }: Props): JSX.Element {
+  const makeCancelable = useCancelablePromise();
+  const {
+    electionDefinition,
+    auth,
+    logger,
+    hasPrinterAttached,
+    printer,
+    addPrintedBallot,
+  } = useContext(AppContext);
+  assert(electionDefinition);
+  assert(isAdminAuth(auth) || isSuperadminAuth(auth));
+  const userRole = auth.user.role;
+  const { election, electionHash } = electionDefinition;
+
+  const needsPrinter = window.kiosk && !hasPrinterAttached;
+  const initialState: ModalState = needsPrinter ? 'no-printer' : 'options';
+  const printLock = useLock();
+
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [modalState, setModalState] = useState<ModalState>(initialState);
+  const [printFailed, setPrintFailed] = useState(false);
+
+  const [ballotMode, setBallotMode] = useState(
+    draftMode ? BallotMode.Draft : BallotMode.Official
+  );
+  const [isAbsentee, setIsAbsentee] = useState(true);
+  const [ballotCopies, setBallotCopies] = useState(1);
+
+  const [ballotIndex, setBallotIndex] = useState<number>();
+  const ballotStyles = useMemo<BallotStyleData[]>(
+    () =>
+      sortBallotStyleDataByPrecinct(election, getBallotStylesData(election)),
+    [election]
+  );
+
+  useEffect(() => {
+    if (draftMode) {
+      setBallotMode(BallotMode.Draft);
+    } else {
+      setBallotMode(BallotMode.Official);
+    }
+  }, [draftMode]);
```

Comment:
I believe this is to reset the ballot mode between closes and re-opens of the modal, is that right? If so, might make sense to just reset the ballot mode in `closeModal` instead. (I may be missing something here!)

Reply:
When you close the modal in admin mode, you don't know whether the next time it will be opened in admin or superadmin mode. So I don't think it can be reset in `closeModal`?

The purpose is not to reset between closes, but to reset between userRole switches.

Reply:
Got it! The current approach makes sense then

In the case of a user role switch, I imagine that this component would completely unmount and state would be cleared, making this `useEffect` unnecessary. (I could be wrong here.) But it also doesn't hurt to have it

Reply:
I hit that issue and thought I needed it, but I think that was because of a different issue. I tested and it is unnecessary on latest, so I'll remove it!